query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapTestDirectorySettingsSpec.groovy

```diff
@@ -392,4 +392,38 @@ class LdapTestDirectorySettingsSpec extends Specification {
         report.result().getValue()
         ldapConnectionIsClosed
     }
+
+    def 'When useCase = AttributeStore, checkGroupObjectClass and checkGroup should be applied'() {
+        setup:
+        def ldapSettings = initLdapSettings(AUTHENTICATION, true)
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME)       : noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)          : simpleBindInfo().getValue(),
+                (LdapDirectorySettingsField.DEFAULT_FIELD_NAME): ldapSettings.getValue()]
+        action.setValue(args)
+        def utilsMock = Spy(LdapTestConnectionSpec.LdapTestingUtilsMock)
+        action.setTestingUtils(utilsMock)
+
+        when:
+        FunctionReport report = action.getValue()
+
+        then:
+        3 * utilsMock.getLdapQueryResults(*_)
```

Comment:
I would say both versions are fragile. Both leak internal details, and listing out exact LDAP queries seems to leak a little more to me.